modules/misc/xdg-user-dirs.nix

```diff
@@ -0,0 +1,109 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.xdg.userDirs;
+
+in {
+  meta.maintainers = with maintainers; [ pacien ];
+
+  options.xdg.userDirs = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to manage <filename>$XDG_CONFIG_HOME/user-dirs.dirs</filename>.
+        This option requires <option>xdg.enable</option> to be enabled.
+        </para>
+        <para>
+        The generated file is read-only.
+      '';
+    };
+
+    # Well-known directory list from
+    # https://gitlab.freedesktop.org/xdg/xdg-user-dirs/blob/master/man/user-dirs.dirs.xml
+
+    desktop = mkOption {
+      type = types.str;
+      default = "$HOME/Desktop";
+      description = "The Desktop directory.";
+    };
+
+    documents = mkOption {
+      type = types.str;
+      default = "$HOME/Documents";
+      description = "The Documents directory.";
+    };
+
+    download = mkOption {
+      type = types.str;
+      default = "$HOME/Downloads";
+      description = "The Downloads directory.";
+    };
+
+    music = mkOption {
+      type = types.str;
+      default = "$HOME/Music";
+      description = "The Music directory.";
+    };
+
+    pictures = mkOption {
+      type = types.str;
+      default = "$HOME/Pictures";
+      description = "The Pictures directory.";
+    };
+
+    publishShare = mkOption {
+      type = types.str;
+      default = "$HOME/Public";
+      description = "The Public share directory.";
+    };
+
+    templates = mkOption {
+      type = types.str;
+      default = "$HOME/Templates";
+      description = "The Templates directory.";
+    };
+
+    videos = mkOption {
+      type = types.str;
+      default = "$HOME/Videos";
+      description = "The Videos directory.";
+    };
+
+    extraDirs = mkOption {
+      type = with types; attrsOf str;
+      default = { };
+      example = literalExample ''
+        {
+          XDG_MISC_DIR = "$HOME/Misc";
+        }
+      '';
+      description = "Other user directories.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = config.xdg.enable;
+        message = "xdg.enable is required to use xdg.userDirs options.";
```

Comment:
I'm unsure whether this assertion is necessary.
Other modules seem to set some `xdg.configFile.abc` even when `xdg.enable` is `false`.

Reply:
No, I think you can remove this and the "This option requires <option>xdg.enable</option> to be enabled." line above.

Reply:
Assertion removed.